### DISCUSSION TITLE
Filtering datapoints by frame_id in the header of ROS message

### DISCUSSION THREAD

**Discussion by isysoev** [facontidavide/PlotJuggler#852] [Q&A]
Hello everyone,

I'm currently struggling to plot a seemingly basic ROS2 topic with PlotJuggler, and I'm wondering if I'm missing something, or perhaps doing things in a not-very-ROS way.

I have a system that tracks a number of objects and publishes their positions into the topic /positions. To disambiguate which position is coming from which object, I use the type Vector3Stamped, and record object ids as frame_id in the header of the message.

When I try to plot this in PlotJuggler, it plots positions of all objects at once, resulting in a jagged plot. I would love to filter the incoming messages by frame_id and only display those belonging to a particular object. I tried to accomplish this using custom time series and writing filtering function in Lua, but it seems that it is not possible to create one when a string field (in this case, frame_id) is selected for one of the additional timeseries.

I wonder if there is another way to accomplish this? Or is it a bad ROS2 practice to publish all positions in one topic, and would the standard practice be using separate topics like /positions/<object_id>?

**Comment by lukestroh** [comment]
I would also like this feature (or be told to not do it).
